src/java/org/apache/cassandra/index/sai/metrics/ColumnQueryMetrics.java

```diff
@@ -97,6 +104,14 @@ public BalancedTreeIndexMetrics(IndexIdentifier indexIdentifier)
             postingsListener = new PostingListEventsMetrics(postingDecodes);
         }
 
+        @Override
+        public void release()
+        {
+            super.release();
+            Metrics.remove(createMetricName("NumPostings", BALANCED_TREE_POSTINGS_TYPE));
+            Metrics.remove(createMetricName("PostingDecodes", BALANCED_TREE_POSTINGS_TYPE));
```

Comment:
nit: We could make constants out of "PostingDecodes" and "NumPostings" now, I guess, since we use them multiple places, but not a big deal either way.